### ISSUE TITLE
change react-intl language triggers page component to rerender

### ISSUE THREAD

**Issue by petitkriket** [react-boilerplate/react-boilerplate#2809]
Doing a personnal project and cannot figure why changing of language rerenders the current page component which reset children components.

Is it related to the router and switch ? Language are not hot reloaded like in react-i18next ?

## Description
Changing language rerender current page component and it's childs

## Steps to reproduce
Change language using language selector

**Expected behavior**
Changing a language  rerenders only texts and concerned items

**Screenshots**
![Peek 20-11-2019 18-55](https://user-images.githubusercontent.com/1753019/69264364-78b8dd00-0bc7-11ea-8fc7-7796c03c22c6.gif)

## Versions

- React-Boilerplate: v4.0.0
- Node/NPM:
- Browser:

**Comment by julienben** [comment]
You mention hot reloading so I assume you're talking about development right? If yes, check out PR #2810. We're trying to fix issues with hot reloading there. Can you try to apply these updates to your app and see if it helps?

**Comment by petitkriket** [comment]
Thank you for your quick reply.

Sorry for the lack of clarity, I am very new to React. 

It's not only about dev environement. I would like the strings to update without re-rendering the whole current displayed component on language update but only the translated parts (strings) as it did with react-i18next + create-react-app in a previous attempt.

Is it possible with react-intl or should I use react-i18next instead ?

**Comment by Can-Sahin** [comment]
Can you send the console logs in the browser ? 

When you change the json file console logs should be printed starting with this text
```
[HMR] Checking for updates on the server...
```

**Comment by julienben** [comment]
@Can-Sahin I don't think that's relevant. It's not a code change so there's no reason for HMR to be visible in the console.

The issue seems to be with changing the locale using the LocaleToggle. Whether in dev or production, the entire page rerenders. I see the same thing in the sample app, both in `dev` and `master` so it's not something we broke recently.

**Comment by Can-Sahin** [comment]
Sorry when I read the `Language are not hot reloaded like in react-i18next ?` section. I thought its a hot loading problem. It's actually not. 

I have a quick suggestion can you try deleting `key={locale}` in `LanguageProvider` Component in `containers` folder. See if it works ? 

It should look like this without the key props

```javascript
export function LanguageProvider(props) {
  return (
    <IntlProvider
      locale={props.locale}
      messages={props.messages[props.locale]}
    >
      {React.Children.only(props.children)}
    </IntlProvider>
  );
}

```

@julienben also can you try that too. What is the exact reason of key in props there. It causes the re-rendering. Removing it makes the language transition smoother. But I might be missing an edge case maybe forced that key props to be there ??

**Comment by petitkriket** [comment]
My bad for the vocab if it's confusing. I removed the key . As expected It breaks the component with  
```
react.development.js?72d0:188 Warning: Each child in a list should have a unique "key" prop.
```
It's not exactly a bug since I believe it works as expected once you uninstall examples, but the rerender is still a bit unexpected.

**Comment by Can-Sahin** [comment]
In the master branch yeah just checked it isnt working without the key prop but I was working #2810 branch and it was working without re-rendering there. So there are quite alot of changes since the master release. I cannot possibly know why with current info. If you have a repository of your current app I can download and take a look at there.

**Comment by petitkriket** [comment]
owww ! It's very kind of you !
 the repository is located [here](https://github.com/petitkriket/lespassagers-client)
Thanks for the information ! Basically if I checkout the twenty files your branch I have some chances to make it work more smoothly ?

Is there any change so that your PR makes it to the next release ?

**Comment by Can-Sahin** [comment]
Hmm I honestly have no idea. There are so many errors happening due to missing tokens and missing localhost api's. I also highly suggest if you are starting the project new just start from the dev branch of this repo. Master is kinda old.  React component classes and lifecycle hooks getting old-fashioned and left behind. Would really suggest functional components with hooks. 

`Thanks for the information ! Basically if I checkout the twenty files your branch I have some chances to make it work more smoothly ?`

Well not exactly because it is based on dev branch. Its really far ahead of master. In my branch of fixing hot loader issues. When I delete `key={locale}` props its working in the demo app. Here in your repo it is not. So no help from me :(

**Comment by julienben** [comment]
@Can-Sahin is totally right, removing the key fixes it in the demo app. (I submitted it as a fix in PR #2818.)

@petitkriket You did that same thing and got the `Each child in a list should have a unique "key" prop.` warning? That's extremely strange as the LanguageProvider does not render a list. My best advice for you is to go back in time, commit by commit, and see at which point the issue was introduced into your repo. (Make sure you remove the key when testing each commit.)

**Comment by petitkriket** [comment]
@julienben will do, thanks !
